app/controllers/file_controller.rb

```diff
@@ -1,16 +1,29 @@
-class FileController < ApplicationController
-  before_filter :require_user
-  before_filter :redirect_to_latest_version
-  before_filter :load_file
+require 'httpclient'
+require 'json'
 
+class FileController < ApplicationController
   before_filter do
+    require_user
+  end
+
+  # Do not force redirect to latest version for key lookup
+  before_filter(except: %i[storage_key]) do
+    redirect_to_latest_version
+  end
+
+  # Do not force load of file for key lookup
+  before_filter(except: %i[storage_key]) do
+    load_file
+  end
+
+  before_filter(except: %i[storage_key]) do
     unless current_user_can_download?(@file.inv_version.inv_object)
       flash[:error] = 'You do not have download permissions.'
       render file: "#{Rails.root}/public/401.html", status: 401, layout: false
     end
   end
 
-  before_filter(only: [:download]) do
+  before_filter(only: %i[download presign]) do
```

Comment:
this is really just a preference, no need to change what you have here. This is a more common approach:
```ruby
before_filter :check_version, only: %i[download pressing]

...
private

def check_version
  version = @file.inv_version
  ...
end
```